spec/ruby/core/io/timeout_spec.rb

```diff
@@ -0,0 +1,104 @@
+# -*- encoding: utf-8 -*-
+require_relative '../../spec_helper'
+
+describe "IO#timeout" do
+  before :each do
+    @fname = tmp("io_timeout.txt")
+    @file = File.open(@fname, "a+")
+
+    @rpipe, @wpipe = IO.pipe
+    # There is no strict long term standard for pipe limits (2**16 bytes currently). This is an attempt to set a safe
+    # enough size to test a full pipe.
+    @more_than_pipe_limit = 1 << 18
+  end
+
+  after :each do
+    @rpipe.close
+    @wpipe.close
+
+    @file.close
+    rm_r @fname
+  end
+
+  ruby_version_is "3.2" do
+    it "files have timeout attribute" do
+      @fname = tmp("io_timeout_attribute.txt")
+      touch(@fname)
+
+      @file.timeout.should == nil
+
+      @file.timeout = 1.23
+      @file.timeout.should == 1.23
+    end
+
+    it "IO instances have timeout attribute" do
+      @rpipe.timeout.should == nil
+      @wpipe.timeout.should == nil
+
+      @rpipe.timeout = 1.23
+      @wpipe.timeout = 4.56
+
+      @rpipe.timeout.should == 1.23
+      @wpipe.timeout.should == 4.56
+    end
+
+    it "raises IO::TimeoutError when timeout is exceeded for .read" do
+      @rpipe.timeout = 0.01
```

Comment:
I think we should have the timeout even lower, like `0.001`.
Or maybe `0` works?
The problem is if the operation would complete in that short time then the spec would fail.

Reply:
I changed it to 0.001 and added tests for 0. The reason is 0 only tests the "last attempt" case, without polling, so I kept the rest on 0.001.